pkg/virt-controller/watch/migration.go

```diff
@@ -222,19 +224,23 @@ func (c *MigrationController) runWorker() {
 }
 
 func (c *MigrationController) Execute() bool {
-	key, quit := c.Queue.Get()
+	queue := &c.Queue
+	if c.Queue.Len() == 0 && c.pendingQueue.Len() > 0 {
```

Comment:
So this prioritize the `queue` but can cause that a multiple workers can be stuck on the `pendingQueue`.
Maybe we don't want to do it. Maybe simple stack that pops when a migration is finished would be sufficient.

Reply:
Having 2 queues is scary I agree, but we need a way to keep the main queue short, or we'll keep starving active migrations when we have a lot in pending.
So far, this is the only thing I could come up with that doesn't fundamentally change how we do controllers...
Keep in mind this is a draft, I opened it mostly to see how many tests would blow up (none it seems!), I have not yet committed to this approach.
I'm not sure what you mean by `simple stack that pops when a migration is finished`, is that describing current behavior or is it a new idea to fix the issue at hand?
Thanks!

Reply:
If by "sleep" you mean "not re-enqueue them" then yes, anything that keeps them out of the queue will do.
We could keep that secondary queue and move 1 item from it to the main queue when a migration finishes, however that scares more than the current state of this PR...